src/useStream.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef, useState } from 'react';
+import { ConduitClient, type ConduitClientConfig } from './client';
+
+export function useStream<T>(config: ConduitClientConfig<T>, enabled: boolean) {
+  const clientRef = useRef<ConduitClient<T> | null>(null);
+  const isConnectingRef = useRef(false);
+  const [isConnected, setIsConnected] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [messages, setMessages] = useState<T[]>([]);
```

Comment:
Let's not store the history of messages, we should leave it up to the caller to assign a `onMessage` and decide what to do with messages